backend/tests/conftest.py

```
@@ -32,79 +37,235 @@
+
+            if is_healthy:
+                logger.info(f"All services healthy after {attempt} attempts")
+                return True, ""
+
+            logger.info(
+                f"Services not ready (attempt {attempt}/{max_attempts}): {message}"
+            )
+
+            if attempt < max_attempts:
+                logger.info(f"Waiting {poll_interval} seconds before next check...")
+                time.sleep(poll_interval)
```

Ruff (0.8.2)

Undefined name `time`

(F821)